File: Java/53.Maximum_Subarray.java
class Solution {
    public int maxSubArray(int[] nums) {
        int total = nums[0];
        int totalSoFar = total;

        for(int i = 1; i < nums.length; i++) {
            totalSoFar = Math.max(totalSoFar + nums[i], nums[i]);
            total = Math.max(total, totalSoFar);
        }
        return total;
    }
}

File: Python3/53.Maximum_Subarray.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        total = nums[0] 
        total_so_far = total

        for i in range(len(nums) - 1):
            total_so_far = max(total_so_far + nums[i + 1], nums[i + 1])
            total = max(total, total_so_far)
        return total